### PR TITLE
Alerting: Add support for TTL for pushover for Mimir Alertmanager

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

This PR adds support for the `ttl` configuration option for `Pushover` when configuring an external Mimir Alertmanager.

This is the front-end counterpart to https://github.com/grafana/mimir/pull/6560

**Who is this feature for?**

Users of Mimir or Cortext Alertmanager.

**Notes for reviewers**

This currently requires an unreleased version of Mimir – but it should be released and deployed by the time Grafana 10.3 lands.

### CODE REVIEW

File: docs/sources/old-alerting/notifications.md [hunk 1]
```diff
@@ -183,6 +183,7 @@ To set up Pushover, you must provide a user key and an API token. Refer to [What
 | OK priority    | The priority OK notifications are sent; if not set, then OK notifications are sent with the priority set for alerting notifications |
 | Retry          | How often (in seconds) the Pushover servers send the same notification to the user. (minimum 30 seconds)                            |
 | Expire         | How many seconds your notification will continue to be retried for (maximum 86400 seconds)                                          |
+| TTL            | A number of seconds that the message will live, before being deleted automatically                                                  |
 | Alerting sound | The sound for alerting notifications                                                                                                |
```
Comment (brendamuir) [inline docs/sources/old-alerting/notifications.md:187]:
```suggestion
| TTL            | The number of seconds before a message expires and is deleted automatically                                                  |
| Alerting sound | The sound for alerting notifications                                                                                                |
```

File: public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts [hunk 2]
```diff
@@ -175,6 +175,9 @@ export const cloudNotifierTypes: Array<NotifierDTO<CloudNotifierType>> = [
           placeholder: '1h',
         }
       ),
+      option('ttl', 'TTL', 'A number of seconds that the message will live, before being deleted automatically', {
+        validationRule: '^\\d+$|^$',
```
Comment (brendamuir) [inline public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts:179]: Same as above - what do you think?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -462,6 +462,7 @@ cypress.config.js @grafana/grafana-frontend-platform
 /public/app/routes/ @grafana/grafana-frontend-platform
 /public/app/store/ @grafana/grafana-frontend-platform
 /public/app/types/ @grafana/grafana-frontend-platform
+/public/app/types/alerting.ts @grafana/alerting-frontend
```
Comment (gillesdemey) [inline .github/CODEOWNERS:465]: Sorry for the ping folks, probably makes sense for @grafana/alerting-frontend to own this

File: public/app/types/alerting.ts [hunk 4]
```diff
@@ -144,7 +144,7 @@ export interface NotificationChannelOption {
   validationRule: string;
   subformOptions?: NotificationChannelOption[];
   dependsOn: string;
-  setValueAs?: (value: string | boolean) => string | number | boolean | null;
+  setValueAs?: (value: string | boolean) => string | number | boolean | null | undefined;
```
Comment (soniaAguilarPeiron) [inline public/app/types/alerting.ts:147]: why do we need to add this change?

File: docs/sources/old-alerting/notifications.md [hunk 5]
```diff
@@ -183,6 +183,7 @@ To set up Pushover, you must provide a user key and an API token. Refer to [What
 | OK priority    | The priority OK notifications are sent; if not set, then OK notifications are sent with the priority set for alerting notifications |
 | Retry          | How often (in seconds) the Pushover servers send the same notification to the user. (minimum 30 seconds)                            |
 | Expire         | How many seconds your notification will continue to be retried for (maximum 86400 seconds)                                          |
+| TTL            | A number of seconds that the message will live, before being deleted automatically. Examples: 10s, 5m30s, 8h.                       |
 | Alerting sound | The sound for alerting notifications                                                                                                |
```
Comment (brendamuir) [inline docs/sources/old-alerting/notifications.md:187]: Change this message to be the same as the one below - the number of seconds...

File: public/app/types/alerting.ts (hunk 4 above)
Comment (gillesdemey) [reply public/app/types/alerting.ts:147]: I tried a different approach here first with `setValueAs()` returning `undefined` for validation but `validationRule` worked better here, should revert this.

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM!

**Pull request merged**